packages/code-editor/src/MonacoEditor/modelSessionManager/index.js

```diff
@@ -185,14 +185,17 @@ class ModelSessionManager {
     await this.projectManager.saveFile(filePath, this.sessions[filePath].value)
     this._editorContainer.fileSaved(filePath)
     this.sessions[filePath].saved = true
+    formatDocumentStatus && await this.saveFile(filePath)
   }
 
   async saveCurrentFile() {
     if (!this.currentFilePath) {
       throw new Error('No current file open.')
     }
+    let formatDocumentStatus = false
+    this.projectManager.projectSettings?.settings?.formatSolidity && this.currentFilePath.endsWith('.sol') && this._editor?.getAction('editor.action.formatDocument').run() && (formatDocumentStatus = true)
```

Comment:
what if there have more conditions? the code should be readable

Reply:
ok